src/lang/db/semantic.rs

```diff
@@ -362,9 +371,11 @@ fn lookup_binding<'db>(
 fn file_and_subfiles_with_corresponding_modules<'db>(
     db: &'db dyn Database,
     file: FileId<'db>,
-) -> Option<(HashSet<FileId<'db>>, HashSet<ModuleId<'db>>)> {
-    let mut modules: HashSet<_> = db.file_modules(file).ok()?.iter().copied().collect();
-    let mut files = HashSet::from([file]);
+) -> (OrderedHashSet<FileId<'db>>, OrderedHashSet<ModuleId<'db>>) {
+    let file_modules = db.file_modules(file).expect("file must be OnDisk");
+
+    let mut modules: OrderedHashSet<_> = file_modules.iter().copied().collect();
+    let mut files = OrderedHashSet::from_iter([file]);
```

Comment:
naming it `result_files` would make the below code more readable imho, as there's a lot of collections of files down there